src/js_native_api_v8.cc

```diff
@@ -804,14 +804,14 @@ napi_status NAPI_CDECL napi_create_function(napi_env env,
 }
 
 napi_status NAPI_CDECL
-napi_define_class(napi_env env,
-                  const char* utf8name,
-                  size_t length,
-                  napi_callback constructor,
-                  void* callback_data,
-                  size_t property_count,
-                  const napi_property_descriptor* properties,
-                  napi_value* result) {
+napi_declare_class(napi_env env,
+                   const char* utf8name,
+                   size_t length,
+                   napi_callback constructor,
+                   void* callback_data,
+                   size_t property_count,
+                   const napi_property_descriptor* properties,
+                   napi_value* result) {
```

Comment:
Yes, I did hesitate in fact - since it is not a JS value but it is a `Local` reference.

I went through all the discussions about this and the main reason it was not done is that this is something that is supposed to be easily extended to other JS engines, so maybe some abstraction is to be implemented in Node itself - like pretending that it is a JS value, but then disallowing assignments or casts?

Reply:
> Yes, I did hesitate in fact - since it is not a JS value but it is a `Local` reference.

`Local` only means that it’s something managed through the V8 GC, not that it’s something that you can actually interact with.

> I went through all the discussions about this and the main reason it was not done is that this is something that is supposed to be easily extended to other JS engines, so maybe some abstraction is to be implemented in Node itself -

Why not just use another type that is like `napi_value`, but not the same and not convertible?

> like pretending that it is a JS value, but then disallowing assignments or casts?

That sounds way more complex than using different types. (But it is important that Node-API users don’t start passing this value to other places that are expecting JS values.)

Reply:
Ok, since it is an opaque type anyway and since currently there will be only one function returning it and only function accepting it as input - we can as well use another type - and simply disregard if it is compatible with anything else or not - since it won't be used for anything else. The less we define, the more room there will be to adapt it to another engine.
Normally, with another engine, either this value will be the same as `Function` - which should be ok - or it will be another incompatible proprietary type. So using a different type should cover both cases.